Src/FluentAssertions/Specialized/JsonNodeAssertions.cs

```diff
@@ -0,0 +1,488 @@
+    {
+        CurrentAssertionChain
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is not null)
+            .FailWith("Cannot assert the existence of a property on a <null> JSON node{reason}.")
+            .Then
+            .ForCondition(Subject is not null && Subject[code] is not null)
+            .FailWith("Expected {context:JSON node} to have property {0}{reason}.", code);
+
+        return new AndWhichConstraint<JsonNodeAssertions, JsonNode>(this, Subject[code]);
+    }
+
+    /// <summary>
+    /// Asserts that the current <see cref="JsonNode"/> does not have a property with the specified name.
+    /// </summary>
+    /// <param name="code">The name of the property that should not exist.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<JsonNodeAssertions> NotHaveProperty(string code,
+        [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        CurrentAssertionChain
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is not null)
+            .FailWith("Cannot assert the existence of a property on a <null> JSON node{reason}.")
+            .Then
+            .ForCondition(Subject is null || Subject[code] is null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Did not expect {context:JSON node} to have property {0}{reason}.", code);
+
+        return new AndConstraint<JsonNodeAssertions>(this);
+    }
+
+    /// <summary>
+    /// Asserts that the current <see cref="JsonNode"/> represents a JSON array.
+    /// </summary>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndWhichConstraint<JsonNodeAssertions, IEnumerable<JsonNode>> BeAnArray(
+        [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        CurrentAssertionChain
+            .ForCondition(Subject is JsonArray)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context:JSON node} to be an array{reason}, but {0} is not.", Subject);
+
+        return new AndWhichConstraint<JsonNodeAssertions, IEnumerable<JsonNode>>(this, (Subject as JsonArray)!.AsEnumerable());
```

Comment:
Otherwise we throw an NRE if wrapped inside an `AssertionScope`

```suggestion
        return new AndWhichConstraint<JsonNodeAssertions, IEnumerable<JsonNode>>(this, (Subject as JsonArray)?.AsEnumerable());
```